test/functional/c/test_magnetic_symmetry.cpp

```diff
@@ -421,6 +421,58 @@ TEST(MagneticSymmetry, test_spgms_get_magnetic_dataset_high_mag_symprec) {
     if (dataset != NULL) spg_free_magnetic_dataset(dataset);
 }
 
+TEST(MagneticSymmetry, test_spg_get_magnetic_dataset_non_collinear) {
+    double lattice[3][3] = {{10, 0, 0}, {0, 10, 0}, {0, 0, 10}};
+    double position[][3] = {
+        {0, 0, 0},
+    };
+    int types[] = {1};
+    double spins[3] = {1, 0, 0};
+    int num_atom = 1;
+    SpglibMagneticDataset *dataset;
+    dataset = spg_get_magnetic_dataset(lattice, position, types, spins, 1,
+                                       num_atom, 1, 1e-5);
+
+    ASSERT_TRUE(dataset->n_operations == 16);
+    show_spg_magnetic_dataset(dataset);
+    spg_free_magnetic_dataset(dataset);
+}
+
+TEST(MagneticSymmetry, test_spg_get_symmetry_with_site_tensors_non_collinear) {
+    double lattice[3][3] = {{10, 0, 0}, {0, 10, 0}, {0, 0, 10}};
+    double position[][3] = {
+        {0, 0, 0},
+    };
+    int types[] = {1};
+    double tensors[3] = {1, 0, 0};
+    int num_atoms = 1;
+    int max_size = num_atoms * 96;
```

Comment:
Good question! Max possible number of point group operations is 48 for space group. For magnetic space group, the space is extended by spin degree of freedom, so it should be twice of 48. If all atoms are the lattice points, i.e., if the cell is not a primitive cell, due to lattice translations, the number of symmetry operations in the cell (this is not proper terminology though, allow me) should be multiplied with the number of atoms. Thus, for magnetic space group operations returned by spglib would be at maximum,

48 * 2 * number_of_atoms

This might not be written in documentation.

Reply:
It should be exposed as a function then

Reply:
Do you mean, for example, a function to prepare `size` dependent arrays with the `max_size = 96 * num_atoms`?

Those functions having `max_size` parameter exist by historical reasons. `spg_get_dataset` and `spg_get_magnetic_dataset` can replace those functions most of the cases. So I want to drop functions having `max_size` parameter from support, but quite a few codes may be dependent on those functions.

If we add more API functions, we have to support them. So I want to avoid it as much as possible. I guess most of users use the python interface, for which these functions don't have `max_size`. I don't know what the compromise is.

Reply:
Hmm, I need to check a bit the C and gnu standard if there are feasible ways to report these deprecations. I'll also need to double check why these are not relevant in the `spg_get_dataset` interface.